src/Workspaces/Core/Portable/Workspace/DocumentEventArgs.cs

```diff
@@ -10,6 +11,7 @@ public class DocumentEventArgs : EventArgs
 
         public DocumentEventArgs(Document document)
         {
+            Contract.ThrowIfNull(document);
```

Comment:
[This](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Workspace_Editor.cs,364) can pass in a null argument for a removed document.